src/filelock/_api.py

```diff
@@ -95,11 +95,13 @@ def __init__(
         If this is set to ``False`` then the lock will be reentrant across threads.
         """
         self._is_thread_local = thread_local
+        path = os.fspath(lock_file)
+        path_str = path.decode() if isinstance(path, bytes) else path
```

Comment:
Hmm, with this I'm no longer sure this is something we want to support. The user can put their casting at call side if they must, no?

Reply:
Other parts of the code depend in this being a `str`, like these [debug logging lines](https://github.com/tox-dev/filelock/blob/09ef4128981559ea7a2c4f2c48937d48debd5e63/src/filelock/_api.py#L225) (otherwise we get `b'log lines'`).

So the choice for me was whether to do `filename.decode() if isinstance(filename, bytes) else filename` in multiple spots, or make sure that it's `str` at init time.